docs/device-verification-guide.adoc

```diff
@@ -0,0 +1,1126 @@
+USER_BOARDREV=$(jq -r '.USER_BOARDREV // "unknown"' "$METADATA_FILE")
+echo "  Device MAC: $MAC_ADDR"
+echo "  Board revision: $USER_BOARDREV"
+
+# Cross-reference with manufacturing database
+echo ""
+echo "Cross-referencing with manufacturing database..."
+DB_SECURITY=$(sqlite3 "${RPI_SB_PROVISIONER_MANUFACTURING_DB}" "SELECT pubkey_programmed, devkey_revoked FROM devices WHERE serial = '<device_serial>';" 2>/dev/null || echo "|")
+DB_PUBKEY=$(echo "$DB_SECURITY" | cut -d'|' -f1)
+DB_DEVKEY=$(echo "$DB_SECURITY" | cut -d'|' -f2)
+
+if [ "$DB_PUBKEY" = "1" ] && [ "$DB_DEVKEY" = "1" ]; then
+    echo "✓ Manufacturing database confirms secure boot is properly configured:"
+    echo "  - Customer public key programmed: YES"
+    echo "  - Development key revoked: YES"
+elif [ "$DB_PUBKEY" = "1" ]; then
+    echo "⚠ Manufacturing database shows partial secure boot configuration:"
+    echo "  - Customer public key programmed: YES"
+    echo "  - Development key revoked: NO (secure boot not enforced)"
+else
+    echo "⚠ Manufacturing database shows device was not securely provisioned"
+fi
+----
+
+**Expected Result**: The `CUSTOMER_KEY_HASH` field in the metadata JSON should match the SHA256 hash calculated from your `CUSTOMER_KEY_FILE_PEM`. The metadata will also show additional security information like JTAG lock status, device MAC address, and board revision. This definitively confirms that secure boot is enforced using your specific customer signing key.
+
+=== 2. Firmware Version Verification
+
+==== 2.1 Check Provisioning Configuration
+
+Verify the firmware version used during provisioning:
+
+[source,bash]
+----
+# Check the configuration file for firmware specification
+grep "RPI_DEVICE_FIRMWARE_FILE" /etc/rpi-sb-provisioner/config
+
+# If specified, verify the file exists and note its version
+ls -la $(grep "RPI_DEVICE_FIRMWARE_FILE" /etc/rpi-sb-provisioner/config | cut -d'=' -f2)
+----
+
+==== 2.2 On-Device Firmware Verification
+
+On the provisioned device, check the running firmware version:
+
+[source,bash]
+----
+# Check firmware version (on device) - returns git commit hash
+vcgencmd version
+
+# Check bootloader version and configuration
+sudo rpi-eeprom-update -a  # Shows current and available versions
+
+# Check bootloader configuration in human-readable format
+vcgencmd bootloader_config | strings  # Parse binary output
+
+# Alternative: Check via dmesg
+dmesg | grep -i bootloader
+
+# More detailed bootloader information
+sudo rpi-eeprom-config  # Current bootloader config
+----
+
+**Expected Result**: 
+- `vcgencmd version` returns a git commit hash (e.g., `5560078dcc8591a00f57b9068d13e5544aeef3aa`) rather than a date-based version name. This is normal behavior, and should be verified against the bootloader version you selected during provisioning.
+- `rpi-eeprom-update -a` shows the firmware file used and available updates
+- `vcgencmd bootloader_config | strings` filters the binary output to show readable configuration
```

Comment:
rpi-eeprom-config without arugments is preferred over vcgencmd. We should use the direct device-tree interface wherever possible instead of vcgencmd / vcmailbox